src/useNavigationBuilder.tsx

```diff
@@ -79,7 +84,7 @@ export default function useNavigationBuilder<ScreenOptions extends object>(
   );
 
   const [initialState] = React.useState(() =>
-    router.getInitialState({
+    currentRouter.getInitialState({
       routeNames,
       initialRouteName,
       initialParamsList,
```

Comment:
Let's rename this to `routeParamList`, since it's not really initial as it changes when children change.